### PR TITLE
NamespaceV2 cmdlets for ServiceBus

### PR DESCRIPTION
**Pull request by schaudhari6254888**
## Checklist

- [ ] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/ServiceBus/ServiceBus.Autorest/README.md [hunk 1]
```diff
@@ -100,9 +100,98 @@ directive:
...
+      parameter-name: EncryptionKeyVaultProperty
+    set:
+      parameter-name: KeyVaultProperty
+  - where:
+      model-name: SbNamespace
+      property-name: EncryptionKeyVaultProperty
+    set:
+      property-name: KeyVaultProperty
+
+  - where:
```
Comment (damodaravadhani) [inline src/ServiceBus/ServiceBus.Autorest/README.md:132]: @schaudhari6254888 , Remove IsAutoInflateEnabled as it isn't there for SB

**Comment by damodaravadhani** [comment]
@schaudhari6254888 , push the generated doc templates.

File: src/ServiceBus/ServiceBus.Autorest/custom/Set-AzServiceBusNamespaceV2.ps1 [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+        [Microsoft.Azure.PowerShell.Cmdlets.ServiceBus.Category('Body')]
+        [Microsoft.Azure.PowerShell.Cmdlets.ServiceBus.Support.SkuName]
+        ${SkuName},
+
+        [Parameter(HelpMessage = "The Event Hubs throughput units for Basic or Standard tiers, where value should be 0 to 20 throughput units. The Event Hubs premium units for Premium tier, where value should be 0 to 10 premium units.")]
+        [Microsoft.Azure.PowerShell.Cmdlets.ServiceBus.Category('Body')]
+        [System.Int32]
+        ${SkuCapacity},
+
+        [Parameter(HelpMessage = "Value that indicates whether AutoInflate is enabled for ServiceBus namespace.")]
```
Comment (damodaravadhani) [inline src/ServiceBus/ServiceBus.Autorest/custom/Set-AzServiceBusNamespaceV2.ps1:97]: @schaudhari6254888 , please remove MaximumThroughputUnits and EnableAutoInflate as ServiceBus does not support it.

File: src/ServiceBus/ServiceBus.Autorest/custom/New-AzServiceBusNamespaceV2.ps1 [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+        ${IdentityType},
+
+        [Parameter(HelpMessage = "Properties for User Assigned Identities")]
+        [Microsoft.Azure.PowerShell.Cmdlets.ServiceBus.Category('Body')]
+        [System.Collections.Hashtable]
+        ${UserAssignedIdentity},
+
+        [Parameter(HelpMessage = "The minimum TLS version for the cluster to support, e.g. '1.2'")]
+        [Microsoft.Azure.PowerShell.Cmdlets.ServiceBus.Category('Body')]
+        [Microsoft.Azure.PowerShell.Cmdlets.ServiceBus.Support.TlsVersion]
```
Comment (damodaravadhani) [inline src/ServiceBus/ServiceBus.Autorest/custom/New-AzServiceBusNamespaceV2.ps1:80]: Change to System.String

File: src/ServiceBus/ServiceBus.Autorest/custom/New-AzServiceBusNamespaceV2.ps1 [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ----------------------------------------------------------------------------------
+
+<#
+.Synopsis
+Create the new ServiceBusNamespace Entity.
```
Comment (damodaravadhani) [inline src/ServiceBus/ServiceBus.Autorest/custom/New-AzServiceBusNamespaceV2.ps1:17]: Change this message to "Creates a new ServiceBus namespace"

File: src/ServiceBus/ServiceBus.Autorest/custom/Set-AzServiceBusNamespaceV2.ps1 [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ----------------------------------------------------------------------------------
+
+<#
+.Synopsis
+Update the ServiceBusNamespace Entity
```
Comment (damodaravadhani) [inline src/ServiceBus/ServiceBus.Autorest/custom/Set-AzServiceBusNamespaceV2.ps1:17]: Change this message to "Updates a ServiceBus namespace"

File: src/ServiceBus/ServiceBus.Autorest/custom/Set-AzServiceBusNamespaceV2.ps1 [hunk 6]
```diff
@@ -0,0 +1,237 @@
...
+            $null = $PSBoundParameters.Remove('WhatIf')
+            $null = $PSBoundParameters.Remove('Confirm')
+            
+            $serviceBusNamespace = Get-AzServiceBusNamespaceV2 @PSBoundParameters
+
+            # 2. PUT
+            $null = $PSBoundParameters.Remove('InputObject')
+            $null = $PSBoundParameters.Remove('ResourceGroupName')
+            $null = $PSBoundParameters.Remove('NamespaceName')
+            $null = $PSBoundParameters.Remove('EventHubName')
```
Comment (damodaravadhani) [inline src/ServiceBus/ServiceBus.Autorest/custom/Set-AzServiceBusNamespaceV2.ps1:185]: EventHubName does not exist for this cmdlet. Remove the line.

File: src/ServiceBus/ServiceBus.Autorest/examples/Set-AzServiceBusNamespaceV2.md [hunk 7]
```diff
@@ -0,0 +1,252 @@
...
+                                  }
+TenantId                        :
+Type                            : Microsoft.ServiceBus/Namespaces
+UpdatedAt                       : 11/21/2022 5:01:22 AM
+UserAssignedIdentity            : {
+                                  }
+ZoneRedundant                   : False
+```
+
+#Adds a new KeyVaultProperty to ServiceBus namespace `myNamespace`.
```
Comment (damodaravadhani) [inline src/ServiceBus/ServiceBus.Autorest/examples/Set-AzServiceBusNamespaceV2.md:72]: Please remove "#" before the start of the sentence. Descriptions should not be bold. Please remove # from start of descriptions everywhere, only titles must have descriptions.

File: src/ServiceBus/ServiceBus.Autorest/custom/New-AzServiceBusNamespaceV2.ps1 [hunk 8]
```diff
@@ -0,0 +1,181 @@
...
+        [Parameter(HelpMessage = "This determines if traffic is allowed over public network. By default it is enabled.")]
+        [Microsoft.Azure.PowerShell.Cmdlets.ServiceBus.Category('Body')]
+        [Microsoft.Azure.PowerShell.Cmdlets.ServiceBus.Support.PublicNetworkAccess]
+        ${PublicNetworkAccess},
+
+        [Parameter(HelpMessage = "Name of this SKU.")]
+        [Microsoft.Azure.PowerShell.Cmdlets.ServiceBus.Category('Body')]
+        [Microsoft.Azure.PowerShell.Cmdlets.ServiceBus.Support.SkuName]
+        ${SkuName},
+
```
Comment (damodaravadhani) [inline src/ServiceBus/ServiceBus.Autorest/custom/New-AzServiceBusNamespaceV2.ps1:93]: Please add "ZoneRedundant" parameter to New-Az. It is missing here

File: src/ServiceBus/ServiceBus.Autorest/custom/Set-AzServiceBusNamespaceV2.ps1 [hunk 9]
```diff
@@ -0,0 +1,238 @@
...
+            if ($hasPublicNetworkAccess) {
+                $serviceBusNamespace.PublicNetworkAccess = $PublicNetworkAccess
+            }
+            if ($hasSkuCapacity) {
+                $serviceBusNamespace.SkuCapacity = $SkuCapacity
+            }
+            if ($hasTag) {
+                $serviceBusNamespace.Tag = $Tag
+            }
+            if ($hasDefaultProfile) {
```
Comment (damodaravadhani) [inline src/ServiceBus/ServiceBus.Autorest/custom/Set-AzServiceBusNamespaceV2.ps1:226]: DefaultProfile should not be handled like this. Please check my EventHubs PR for how to handle it.

File: src/ServiceBus/ServiceBus.Autorest/custom/Set-AzServiceBusNamespaceV2.ps1 [hunk 10]
```diff
@@ -177,7 +177,6 @@ function Set-AzServiceBusNamespaceV2{
             $hasAsJob = $PSBoundParameters.Remove('AsJob')
```
Comment (damodaravadhani) [inline src/ServiceBus/ServiceBus.Autorest/custom/Set-AzServiceBusNamespaceV2.ps1:177]: Remove "$hasDefaultProfile = $PSBoundParameters.Remove('DefaultProfile')"

File: src/ServiceBus/ServiceBus.Autorest/custom/New-AzServiceBusNamespaceV2.ps1 [hunk 11]
```diff
@@ -70,10 +70,15 @@ function New-AzServiceBusNamespaceV2{
         [Microsoft.Azure.PowerShell.Cmdlets.ServiceBus.Support.ManagedServiceIdentityType]
         ${IdentityType},
 
+        [Parameter(HelpMessage = "Enabling this property creates a Premium Service Bus Namespace in regions supported availability zones.")]
+        [Microsoft.Azure.PowerShell.Cmdlets.ServiceBus.Category('Body')]
+        [System.Management.Automation.SwitchParameter]
+        ${zoneRedundant},
```
Comment (damodaravadhani) [inline src/ServiceBus/ServiceBus.Autorest/custom/New-AzServiceBusNamespaceV2.ps1:76]: Change to "ZoneRedundant"

File: src/ServiceBus/ServiceBus.Autorest/test/New-AzServiceBusNamespaceV2.Tests.ps1 [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+        $serviceBusNamespace.Name | Should -Be $env.namespaceV4
+        $serviceBusNamespace.IdentityType | Should -Be SystemAssigned
+        $serviceBusNamespace.SkuName | Should -Be Premium
+        $serviceBusNamespace.SkuTier | Should -Be Premium
+        $serviceBusNamespace.Location | Should -Be "East Us"
+
+        # Create namespace with UserAssigned Encryption Enabled
+        $ec1 = New-AzServiceBusKeyVaultPropertiesObject -KeyName key1 -KeyVaulturi $env.keyVaultUri -UserAssignedIdentity $env.msi1
+        $ec2 = New-AzServiceBusKeyVaultPropertiesObject -KeyName key2 -KeyVaulturi $env.keyVaultUri -UserAssignedIdentity $env.msi1
+        $serviceBusNamespace = New-AzServiceBusNamespaceV2 -ResourceGroupName $env.resourceGroup -Name $env.namespaceV5 -SkuName Premium -Location northeurope -IdentityType UserAssigned -UserAssignedIdentityId $env.msi1,$env.msi2 -KeyVaultProperty $ec1,$ec2
```
Comment (damodaravadhani) [inline src/ServiceBus/ServiceBus.Autorest/test/New-AzServiceBusNamespaceV2.Tests.ps1:43]: Can you assert for RequireInfrastructureEncryption false here ?

File: src/ServiceBus/ServiceBus.Autorest/test/Get-AzServiceBusNamespaceV2.Tests.ps1 [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+    It 'Get' {
+        $namespace = Get-AzServiceBusNamespaceV2 -ResourceGroupName $env.resourceGroup -Name $env.namespace
+        $namespace.Name | Should -Be $env.namespace
+    }
+
+    It 'GetViaIdentity' {
+        $namespace = Get-AzServiceBusNamespaceV2 -ResourceGroupName $env.resourceGroup -Name $env.namespace
+        $namespace = Get-AzServiceBusNamespaceV2 -InputObject $namespace
+        $namespace.Name | Should -Be $env.namespace
+    }
```
Comment (damodaravadhani) [inline src/ServiceBus/ServiceBus.Autorest/test/Get-AzServiceBusNamespaceV2.Tests.ps1:27]: Missing testcases for "List" and "List1" parameter sets

**Comment by BethanyZhou** [comment]
please check following signiture issue whether is false positive,
```
 "Module","ClassName","Target","Severity","ProblemId","Description","Remediation"
  "Az.ServiceBus","New-AzServiceBusKeyVaultPropertiesObject","New-AzServiceBusKeyVaultPropertiesObject","1","8100","New-AzServiceBusKeyVaultPropertiesObject Does not support ShouldProcess but the cmdlet verb New indicates that it should.","Determine if the cmdlet should implement ShouldProcess and if so determine if it should implement Force / ShouldContinue"
  "Az.ServiceBus","New-AzServiceBusNamespaceV2","New-AzServiceBusNamespaceV2","1","8100","New-AzServiceBusNamespaceV2 Does not support ShouldProcess but the cmdlet verb New indicates that it should.","Determine if the cmdlet should implement ShouldProcess and if so determine if it should implement Force / ShouldContinue"
  "Az.ServiceBus","Set-AzServiceBusNamespaceV2","Set-AzServiceBusNamespaceV2","1","8100","Set-AzServiceBusNamespaceV2 Does not support ShouldProcess but the cmdlet verb Set indicates that it should.","Determine if the cmdlet should implement ShouldProcess and if so determine if it should implement Force / ShouldContinue"
  ```
if yes, please go to Exceptions/ServiceBus/signatureissues.cvs to suppress them. 
if no, please resolve these issues.

**Comment by BethanyZhou** [comment]
Hi @Nickcandy , please help review this PR

File: src/ServiceBus/ServiceBus.Autorest/examples/New-AzServiceBusKeyVaultPropertiesObject.md [hunk 14]
```diff
@@ -1,22 +1,8 @@
-### Example 1: {{ Add title here }}
+### Example 1: Construct an in-memory KeyVaultProperties object
 ```powershell
-{{ Add code here }}
+New-AzServiceBusKeyVaultPropertiesObject -KeyName key1 -KeyVaultUri https://{keyVaultName}.vault.azure.net/
 ```
```
Comment (BethanyZhou) [inline src/ServiceBus/ServiceBus.Autorest/examples/New-AzServiceBusKeyVaultPropertiesObject.md:4]: Please add output for example

File: tools/StaticAnalysis/Exceptions/Az.ServiceBus/SignatureIssues.csv [hunk 15]
```diff
@@ -50,3 +50,6 @@
...
+"Az.ServiceBus","New-AzServiceBusKeyVaultPropertiesObject","New-AzServiceBusKeyVaultPropertiesObject","1","8100","New-AzServiceBusKeyVaultPropertiesObject Does not support ShouldProcess but the cmdlet verb New indicates that it should.","Determine if the cmdlet should implement ShouldProcess and if so determine if it should implement Force / ShouldContinue"
+"Az.ServiceBus","New-AzServiceBusNamespaceV2","New-AzServiceBusNamespaceV2","1","8100","New-AzServiceBusNamespaceV2 Does not support ShouldProcess but the cmdlet verb New indicates that it should.","Determine if the cmdlet should implement ShouldProcess and if so determine if it should implement Force / ShouldContinue"
+"Az.ServiceBus","Set-AzServiceBusNamespaceV2","Set-AzServiceBusNamespaceV2","1","8100","Set-AzServiceBusNamespaceV2 Does not support ShouldProcess but the cmdlet verb Set indicates that it should.","Determine if the cmdlet should implement ShouldProcess and if so determine if it should implement Force / ShouldContinue"
```
Comment (BethanyZhou) [inline tools/StaticAnalysis/Exceptions/Az.ServiceBus/SignatureIssues.csv:55]:
```suggestion
```

File: src/ServiceBus/ServiceBus.Autorest/custom/New-AzServiceBusNamespaceV2.ps1 [hunk 16]
```diff
@@ -21,7 +21,7 @@ Creates a new ServiceBus namespace.
 
 function New-AzServiceBusNamespaceV2{
 	[OutputType([Microsoft.Azure.PowerShell.Cmdlets.ServiceBus.Models.Api202201Preview.ISbNamespace])]
-    [CmdletBinding(PositionalBinding = $false, ConfirmImpact = 'Medium')]
+    [CmdletBinding(PositionalBinding = $false,SupportsShouldProcess, ConfirmImpact = 'Medium')]
```
Comment (BethanyZhou) [inline src/ServiceBus/ServiceBus.Autorest/custom/New-AzServiceBusNamespaceV2.ps1:24]: Correct indents in scripts please

File: src/ServiceBus/ServiceBus.Autorest/custom/Set-AzServiceBusNamespaceV2.ps1 [hunk 17]
```diff
@@ -21,7 +21,7 @@ Updates a ServiceBus namespace
 
 function Set-AzServiceBusNamespaceV2{
 	[OutputType([Microsoft.Azure.PowerShell.Cmdlets.ServiceBus.Models.Api202201Preview.ISbNamespace])]
-    [CmdletBinding(DefaultParameterSetName = 'SetExpanded', PositionalBinding = $false, ConfirmImpact = 'Medium')]
+    [CmdletBinding(DefaultParameterSetName = 'SetExpanded', PositionalBinding = $false,SupportsShouldProcess, ConfirmImpact = 'Medium')]
```
Comment (BethanyZhou) [inline src/ServiceBus/ServiceBus.Autorest/custom/Set-AzServiceBusNamespaceV2.ps1:24]: Correct indents in scripts please

**Review by BethanyZhou — CHANGES_REQUESTED** [review]

File: tools/StaticAnalysis/Exceptions/Az.ServiceBus/SignatureIssues.csv (hunk 15 above)
Comment (BethanyZhou) [reply tools/StaticAnalysis/Exceptions/Az.ServiceBus/SignatureIssues.csv:55]: Added ShouldProcess, these issues should go

File: src/ServiceBus/ServiceBus.Autorest/11 [hunk 18]
```diff
@@ -0,0 +1 @@
+10
```
Comment (Nickcandy) [inline src/ServiceBus/ServiceBus.Autorest/11:1]: And are these two files generated? 11 and 13

Comment (schaudhari6254888) [reply src/ServiceBus/ServiceBus.Autorest/11]: Removed

Comment (damodaravadhani) [reply src/ServiceBus/ServiceBus.Autorest/11]: yes @Nickcandy , these are generated everytime. Not sure why.

**Review by Nickcandy — APPROVED** [review]

**Review by BethanyZhou — APPROVED** [review]
Looks good to me

**Comment by BethanyZhou** [comment]
Hi @Nickcandy , please help sync these changes to main branch.

**Comment by BethanyZhou** [comment]
@schaudhari6254888 @damodaravadhani could you please provide change logs?

**Comment by damodaravadhani** [comment]
* Providing for NamespaceV2 cmdlets for ServiceBus. The existing namespace cmdlets will be configured to NamespaceV2 cmdlets in the next breaking change release. We highly recommend that you use NamespaceV2 cmdlets for ServiceBus.

**Pull request merged**